### ISSUE TITLE
Patching does not add new elements

### ISSUE THREAD

**Issue by harmonwood** [phpro/soap-client#4]
When adding a new element to a WSDL node and then run the generate:types command again it does not get added as a class attribute via an `<xsd:import schemaLocation ..../>` file.
WSDL was:

    ....
    <xs:complexType name="Field">
        <xs:sequence>
            <xs:element minOccurs="0" name="Description" nillable="true" type="xs:string"/>
            <xs:element minOccurs="0" name="FieldType" nillable="true" type="xs:string"/>
        </xs:sequence>
    </xs:complexType>
    ....

    ....
    <xs:complexType name="Field">
        <xs:sequence>
            <xs:element minOccurs="0" name="Description" nillable="true" type="xs:string"/>
            <xs:element minOccurs="0" name="FieldType" nillable="true" type="xs:string"/>
            <xs:element minOccurs="0" name="SortOrder" type="xs:int"/>
        </xs:sequence>
    </xs:complexType>
    ....

 It does create a backup file and state that it is patching like so:

    Type Field exists. Trying to patch ...Patched!

The file and the backup are identical without the new element in the class.

    class Field
    {

       /**
        * @var string
        */
       protected $Description;

       /**
        * @var string
        */
       protected $FieldType;

    }

**Comment by harmonwood** [comment]
The patch file generated is:

    +++ /Field.php▸ 2016-04-20 10:24:44.000000000 -0500
    @@ -46,11 +46,6 @@
         protected $FieldType;
    -
    -    /**
    -     * @var int
    -     */
    -    protected $SortOrder;

**Comment by veewee** [comment]
Hello @harmonwood,

Thanks for reporting the bug. I've added some failing tests but need to investigate how we van solve this issue. I'll keep you posted when there is any progess on the issue!

**Comment by veewee** [comment]
Hi @harmonwood , @Aluok, @janvernieuwe,


I started working on the zend-code featue and got a first POC version going for the type generator:
https://github.com/veewee/soap-client/commit/00da7f41d32a65cf9fdd906da2e86288d200e19f

(note that the class generator is not implemented yet)

These files can be tested with for example:
```
./bin/soap-client generate:types --wsdl="http://wsf.cdyne.com/WeatherWS/Weather.asmx?WSDL" --namespace="Testing" gentest
```

The code works for both existing and new files and patches everything the way it should.
There are still some problems with zend-code like e.g:

- I cant remove a property from a class to overwrite it
- It uses full 'implements' instead of the short link

There are some things that still need to be fixed and a lot of work on tests and documentation. But currently I am pretty satisfied about using zend-code for the code generation.

I also added a RuleSet feature. Currently there is just one hardcoded rule to generate the classes, but in the future you might be able to write your own rule and code assemblers.
Feel free to take a look, test it out and give your feedback.

Thanks
